### ISSUE TITLE
Proxy option documentation issue

### ISSUE THREAD

**Issue by nico29** [gatsbyjs/gatsby#4305]
## Description

Before everything else, awesome job 🤘
I was trying to use the [api proxy](https://www.gatsbyjs.org/docs/api-proxy/#api-proxy) feature to connect to my api running on a local express dev server.
I followed the documentation pattern for the proxy configuration and the URL ended up being incorrect

### Environment

Gatsby version: 1.9.218 | cli version 1.9.218
Node.js version: v9.6.1
Operating System: MacOS 10.13.3

### File contents:

`gatsby-config.js`: 
```javascript
module.exports = {
  proxy: {
    prefix: '/api',
    url: 'http://localhost:3000/api',
  },
};
```
`package.json`:  not changed
`gatsby-node.js`:  not changed
`gatsby-browser.js`:  not changed
`gatsby-ssr.js`:  not changed

### Actual result

If I do `fetch('/api/somepath')` then the request will hit my backend on :3000 with the URL /api/api/somepath. 

![gatsby_url_proxy](https://user-images.githubusercontent.com/3858041/36848150-bb699dbc-1d60-11e8-9bf3-019099c18eec.png)

### Expected behavior

I think this is only a documentation issue, and that the documentation sample

```javascript
module.exports = {
  proxy: {
    prefix: "/api",
    url: "http://dev-mysite.com/api/",
  },
};

fetch('/api/todos'); //http://dev-mysite.com/api/todos
```

should become like follows so that the fetch example behaves as intended:

```javascript
module.exports = {
  proxy: {
    prefix: "/api",
    url: "http://dev-mysite.com",
  },
};
```

If this is indeed a documentation issue and not me misunderstanding the proxy config, I'ld happy to submit a PR to change the documentation.

Thank you 😸

**Comment by jlengstorf** [comment]
This looks like a documentation issue. If you've verified that the updated config works:

```js
module.exports = {
  proxy: {
    prefix: "/api",
    url: "http://dev-mysite.com",
  },
};
```

Please submit a PR. Thanks so much!

**Comment by nico29** [comment]
@jlengstorf can I close this ?

**Comment by jlengstorf** [comment]
Whoops, sorry — closed by #4310.
